input/Inference/test-inference-246.in

```diff
@@ -0,0 +1,364 @@
+             dataset = "GridVelocities_x";
+             file = "input/cosmo_grid_velocities_x.h5";
+             name = "velocity_x";
+             type = "field";
+         }
+         FVY {
+             coords = "tzyx";
+             dataset = "GridVelocities_y";
+             file = "input/cosmo_grid_velocities_y.h5";
+             name = "velocity_y";
+             type = "field";
+         }
+         FVZ {
+             dataset = "GridVelocities_z";
+             file = "input/cosmo_grid_velocities_z.h5";
+             name = "velocity_z";
+             type = "field";
+         }
+         PDX {
+             attribute = "x";
+             coords = "tzyx";
+             dataset = "ParticleDisplacements_x";
+             file = "input/cosmo_particle_displacements_x.h5";
+             name = "dark";
+             type = "particle";
+         }
+         PDY {
+             attribute = "y";
+             coords = "tzyx";
+             dataset = "ParticleDisplacements_y";
+             file = "input/cosmo_particle_displacements_y.h5";
+             name = "dark";
+             type = "particle";
+         }
+         PDZ {
+             attribute = "z";
+             dataset = "ParticleDisplacements_z";
+             file = "input/cosmo_particle_displacements_z.h5";
+             name = "dark";
+             type = "particle";
+         }
+         PVX {
+             attribute = "vx";
+             coords = "tzyx";
+             dataset = "ParticleVelocities_x";
+             file = "input/cosmo_particle_velocities_x.h5";
+             name = "dark";
+             type = "particle";
+         }
+         PVY {
+             attribute = "vy";
+             coords = "tzyx";
+             dataset = "ParticleVelocities_y";
+             file = "input/cosmo_particle_velocities_y.h5";
+             name = "dark";
+             type = "particle";
+         }
+         PVZ {
+             attribute = "vz";
+             dataset = "ParticleVelocities_z";
+             file = "input/cosmo_particle_velocities_z.h5";
+             name = "dark";
+             type = "particle";
+         }
+         file_list = [ "FD", "FVX", "FVY", "FVZ", "PDX", "PDY", "PDZ", "PVX", "PVY", "PVZ" ];
+         format = "music";
+         monitor_iter = 1;
+     }
+     list = [ "hdf5", "cosmology" ];
+     new = true;
+ }
+
+ Mesh {
+     root_blocks = [ 2, 2, 2 ];
+     root_rank = 3;
+     root_size = [ 32, 32, 32 ];
+ }
+ Method {
+     list = [ "ppm", "pm_deposit", "gravity", "pm_update",
+              "comoving_expansion", "inference" ];
+     ppm {
+         courant = 0.5;
+         diffusion = false;
+         dual_energy = true;
+     }
+     gravity {
+         accumulate = true;
+         grav_const = 1.0;
+         order = 2;
+         solver = "dd";
+     }
+     inference {
+         level_base  = 2;
+         level_array = 4;
+         level_infer = 6;
+         array_dims = [ 32, 32, 32 ];
+         array_size = [ 64, 64, 64 ];
```

Comment:
I think(?) it's correct as stated, though the distinction between "array_dims" and "array_size" is confusing, since the "level array" is a chare array, each of whose elements is associated with a collection of "inference arrays". "array_dims" is the size of the inference arrays, whereas "array_size" is the size of the level array. Since neither parameter is actually used except for checking consistency, I may consider removing them altogether.